### PR TITLE
[WIP] Scripts/TheEye: Update Kael'thas Sunstrider's script

### PR DESCRIPTION
**Pull request by ghost**
- Unhacked advisor resurrection
- Corrected combat phase transition timer (engage combat 5 seconds after the last advisor dies rather than a minute and half).
- Corrected wrong Fireball spell ID being used, previously supported by a very ugly hack
- Use proper unit field updates based on sniffs (remove some improper UNIT_FLAGS_NON_ATTACKABLE flags being set)
- Updated to EventMaps
- Unhack gravity lapse
- Several gameobjects and creature spawns missing (relevant to the 50% hp phase transition).

TO DO
- Study the 50% hp phase transition better (there's a myriad of visual and dummy spells being cast...)
- The resurrection spell applies the hp modifier _before_ healing so the advisors are at 50% once revived
- More timer corrections
- Unhack weapon summon spells
- Advisor resurrection spells needs target conditions
- The weapon's AI needs improvements (i.e the staff should cast frost bolts and the bow should attack from afar).

Closes https://github.com/TrinityCore/TrinityCore/issues/4180, https://github.com/TrinityCore/TrinityCore/issues/4181, https://github.com/TrinityCore/TrinityCore/issues/8479,
https://github.com/TrinityCore/TrinityCore/issues/1885

Just opening this so it doesn't end up getting entirely wasted away in case anything happens

### CODE REVIEW

File: src/server/scripts/Outland/TempestKeep/Eye/boss_kaelthas.cpp [hunk 1]
```diff
@@ -456,557 +438,290 @@ class boss_kaelthas : public CreatureScript
...
-                            //ShockBarrier_Timer
-                            if (ShockBarrier_Timer <= diff)
+                            if (events.IsInPhase(PHASE_TRANSITION) || _inGravityLapse)
+                                events.ScheduleEvent(EVENT_SHOCK_BARRIER, _inGravityLapse ? 60000 : 20000);
+                            break;
+                        case EVENT_END_TRANSITION:
+                            DoCast(me, SPELL_EXPLODE);
+                            me->InterruptNonMeleeSpells(false);
+                            me->RemoveAurasDueToSpell(SPELL_FULLPOWER);
+                            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
```
Comment (Nyeriah) [inline src/server/scripts/Outland/TempestKeep/Eye/boss_kaelthas.cpp]: I ended up opening the PR with an older version of the file but anyway I know that it should remove the not selectable flag in this line instead.

**Comment by Killyana** [comment]
According to wowhead the mind control spell is the 36797 and not the 32830, also I never saw a player mind controlled with the actual script.

**Comment by Nyeriah** [comment]
Our current script probably dates from a time where mind control mechanics weren't supported by the core so the spell as commented out. On my rework I enabled it but hadn't looked into that spell in particular yet, thanks for pointing out, as that would one I wouldn't be able to catch on the sniffs...

**Comment by jackpoz** [comment]
please fix the travis build

**Comment by Nyeriah** [comment]
Took care of most things I could take, closing it now

**Pull request closed without merging**